presto-tests/src/main/java/com/facebook/presto/tests/PlanDeterminismChecker.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.sql.planner.LogicalPlanner;
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.PlanPrinter;
+import com.facebook.presto.testing.LocalQueryRunner;
+
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+import static org.testng.Assert.assertEquals;
+
+public class PlanDeterminismChecker
+{
+    private static final int MINIMUM_SUBSEQUENT_SAME_PLANS = 10;
+
+    private final LocalQueryRunner localQueryRunner;
+    private final Function<String, String> planEquivalenceFunction;
```

Comment:
This is convenient for debugging. If I'm in the middle of investigation, it's super-easy to drop in a `plan -> plan.reaplaceAll("_\d+", "")` equivalence function, filtering out all the noise so that I can focus on query plan shape changes